### DISCUSSION TITLE
How to modify SOM_LM-SNNs.py for batch training?

### DISCUSSION THREAD

**Discussion by AXYZdong** [BindsNET/bindsnet#676] [Q&A]
I noticed that `batch_size = 1` in _SOM_LM-SNNs.py_ and if I increase the `batch_size` it will report an error.

**Comment by Hananel-Hazan** [comment]
Although the code can be adapted to different `batch_size`, I dont know if the algorithm learning method can support higher `batch_size`. You are welcome to adapt the code to support higher `batch_size`, let me know how higher `batch_size` effect the learning, this is interesting! I suspect that this kind of change need to change the inhibition mechanism too.

**Reply by AXYZdong** [reply]
I want to increase `batch_size` to fully utilize the GPU. I try to adapt the code to support higher `batch_size`, but I've had no success so far.

**Reply by Hananel-Hazan** [reply]
The update rule and the inhibition signal need to be updated too.
